### PR TITLE
refactor: tier4_map_launch

### PR DESCRIPTION
**Pull request by h-ohta**
## Description

- refactor map.launch.py to use LaunchConfiguration().perform(context) in the future

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by yukkysaito — APPROVED** [review]
LGTM. 

@kenji-miyake 
Pls check too :pray:

File: launch/tier4_map_launch/launch/map.launch.py [hunk 1]
```diff
@@ -114,42 +114,56 @@ def generate_launch_description():
...
-                    container,
-                    map_hash_generator,
-                ]
-            ),
+        launch_arguments
+        + [
+            set_container_executable,
+            set_container_mt_executable,
         ]
+        + [OpaqueFunction(function=launch_setup)]
```
Comment (kenji-miyake) [inline launch/tier4_map_launch/launch/map.launch.py:168]: What for did you have to use `OpaqueFunction`?

Comment (h-ohta) [reply launch/tier4_map_launch/launch/map.launch.py:168]: This is used in the below file.
https://github.com/autowarefoundation/autoware.universe/blob/main/launch/tier4_control_launch/launch/control.launch.py#L385
Actually I copied it.

Comment (kenji-miyake) [reply launch/tier4_map_launch/launch/map.launch.py:168]: The file uses `context`, but this file doesn't.
So, could you remove it from this PR? :pray:

Comment (h-ohta) [reply launch/tier4_map_launch/launch/map.launch.py:168]:
```suggestion
        + [launch_setup]
```
This way?

Comment (kenji-miyake) [reply launch/tier4_map_launch/launch/map.launch.py:168]: No. Please remove `launch_setup` itself.

Comment (h-ohta) [reply launch/tier4_map_launch/launch/map.launch.py:168]: You mean I have to delete `launch_setup` function ?

Comment (kenji-miyake) [reply launch/tier4_map_launch/launch/map.launch.py:168]: Exactly. It's not necessary because it doesn't use the `context` variable.

Comment (h-ohta) [reply launch/tier4_map_launch/launch/map.launch.py:168]: I split https://github.com/autowarefoundation/autoware.universe/pull/952 into two pull requests, refactor and feature.
Now this pull request is refactor. So, I'm going to use `context` variables after merging this to make feature pull request.

Comment (kenji-miyake) [reply launch/tier4_map_launch/launch/map.launch.py:168]: I see. In that case, it's acceptable.

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**